lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -213,20 +215,28 @@ public function fetch($fetchMode = null)
                     $ctorArgs  = isset($args[2]) ? $args[2] : array();
                 }
 
-                $result = db2_fetch_object($this->_stmt);
+                $result = @db2_fetch_object($this->_stmt);
 
                 if ($result instanceof \stdClass) {
                     $result = $this->castObject($result, $className, $ctorArgs);
                 }
-
-                return $result;
+                break;
             case \PDO::FETCH_NUM:
-                return db2_fetch_array($this->_stmt);
+                $result = @db2_fetch_array($this->_stmt);
+                break;
             case \PDO::FETCH_OBJ:
-                return db2_fetch_object($this->_stmt);
+                $result = @db2_fetch_object($this->_stmt);
+                break;
             default:
                 throw new DB2Exception("Given Fetch-Style " . $fetchMode . " is not supported.");
         }
+
+        $error = db2_stmt_errormsg();
+        if ($result === false && $error) {
+            throw new DB2Exception($error);
```

Comment:
Can you add some details here? It is beneficial for consumers to know how we got to this failure (failure during fetch)